R/getPrevalence.R

```diff
@@ -300,17 +346,81 @@ setMethod("getRareTaxa", signature = c(x = "SummarizedExperiment"),
     function(x, rank = NULL, ...){
         # Gets the prevalent taxa
         prev_taxa <- getPrevalentTaxa(x, rank = rank, ...)
+        # If agglomeration was done
         if( !is.null(rank) ){
             # Gets names from specified taxonomic level
             taxa <- rowData(x)[[rank]]
-        } else{
-            # Gets rownames if agglomeration is not done
-            taxa <- rownames(x)
+        } else {
+            # If rownames is not NULL
+            if( !is.null(rownames(x)) ){
+                # Gets rownames 
+                taxa <- rownames(x)
+            } else {
+                # Gets indices of taxa
+                taxa <- 1:nrow(tse)
```

Comment:
I think the check for `is.null(rownames(x))` (`rownames==NULL` is pseudocode and does not work) should be done before the call to `getPrevalentTaxa`. That way the result is always valid and the if tree doesn't get so big.

The rational behind it: If you can avoid putting garbage in, do it. Cleanup afterwards is always messy and bound to fail.

Reply:
Changed